base/checked.jl

```diff
@@ -358,6 +358,21 @@ The overflow protection may impose a perceptible performance penalty.
 """
 checked_cld(x::T, y::T) where {T<:Integer} = cld(x, y) # Base.cld already checks
 
+"""
+    Base.checked_pow(x, y)
+
+Calculates `^(x,y)`, checking for overflow errors where applicable.
+
+The overflow protection may impose a perceptible performance penalty.
+"""
+checked_pow(x::Integer, y::Integer) = checked_power_by_squaring(x, y)
+
+checked_power_by_squaring(x_, p::Integer) =
+    Base.power_by_squaring_with_op(checked_mul, x_, p)
```

Comment:
Then this just becomes:
```suggestion
    Base.power_by_squaring_with_op(x_, p, * = checked_mul)
```

Reply:
nice. done!